pandas/core/arrays/period.py

```diff
@@ -379,38 +378,18 @@ def _add_delta_tdi(self, other):
         delta = self._check_timedeltalike_freq_compat(other)
         return self._addsub_int_array(delta, operator.add)
 
+    @Appender(DatetimeLikeArrayMixin._add_delta.__doc__)
```

Comment:
Why this. Doesn't `PeriodArrayMixin` inherit directly from `DatetimeLikeArrayMixin`? The docstring will be inherited.

Reply:
> The docstring will be inherited.

Will it?  I thought that required some metaclass hijinx

Reply:
```
In [1]: import pandas as pd

In [2]: class A:
   ...:     def a(self):
   ...:         """doc"""
   ...:         pass
   ...:

In [3]: class B(A):
   ...:     pass
   ...:

In [4]: B.a?
Signature: B.a(self)
Docstring: doc
File:      ~/sandbox/pandas/<ipython-input-2-6d9a0c65a8ca>
Type:      function

```

although maybe something else gets in the way?

Reply:
The distinction is that we actually are overriding the method

Reply:
On this branch with

```diff
diff --git a/pandas/core/arrays/period.py b/pandas/core/arrays/period.py
index 1d7b57801..ff297e5bd 100644
--- a/pandas/core/arrays/period.py
+++ b/pandas/core/arrays/period.py
@@ -378,7 +378,6 @@ class PeriodArrayMixin(DatetimeLikeArrayMixin):
         delta = self._check_timedeltalike_freq_compat(other)
         return self._addsub_int_array(delta, operator.add)
 
-    @Appender(DatetimeLikeArrayMixin._add_delta.__doc__)
     def _add_delta(self, other):
         if not isinstance(self.freq, Tick):
             # We cannot add timedelta-like to non-tick PeriodArray
```

the docstring looks fine.

```
In [3]: pd.core.arrays.PeriodArrayMixin._add_delta?
Signature: pd.core.arrays.PeriodArrayMixin._add_delta(self, other)
Docstring:
Add a timedelta-like, Tick or TimedeltaIndex-like object
to self.

Parameters
----------
delta : {timedelta, np.timedelta64, Tick,
         TimedeltaIndex, ndarray[timedelta64]}

Returns
-------
result : same type as self

Notes
-----
The result's name is set outside of _add_delta by the calling
method (__add__ or __sub__), if necessary (i.e. for Indexes).
File:      ~/sandbox/pandas/pandas/core/arrays/period.py
Type:      function

```

Reply:
Ah, hmm, IPython must be doing some magic.

Reply:
Well, for a final check I ran it through sphinx

```
python make.py --single=pandas.core.arrays.PeriodArrayMixin._add_delta

```

and it does whatever IPython is doing. The rendered docstring looks fine to.

In conclusion: I think we can remove the Appender.

Reply:
Is there some cost I'm missing to having `print(PeriodArrayMixin._add_delta.__doc__)` work correctly?

Reply:
Is that a common use case? I assume most people use `help()` or `?`.

Reply:
I usually print the docstring, mostly because I usually forget about the other usages.  This is an internal method so it doesn't _really_ matter.  My view ATM is just that its zero-cost.

Reply:
Same comment here about a single line comment summarizing what it does